pkg/logging/logging.go

```diff
@@ -117,15 +118,19 @@ func (l *Logging) GetLogger() *Logger {
 	return l.logger
 }
 
-// AssertOutput returns an error if output is not a valid logger output.
-func AssertOutput(o string) error {
-	if o == CONSOLE || o == JOURNAL {
-		return nil
-	}
-
-	return invalidOutput(o)
-}
-
-func invalidOutput(o string) error {
-	return fmt.Errorf("%s is not a valid logger output. Must be either %q or %q", o, CONSOLE, JOURNAL)
+// Fatal logs the given error and its stack trace to the console and then exits.
+// This function is to be used if Logging has not yet been created or could not be created:
+//
+//	logs, err := logging.NewLoggingFromConfig(utils.AppName(), logging.Config{})
+//	if err != nil {
+//		logging.Fatal(errors.Wrap(err, "can't configure logging"))
+//	}
+func Fatal(err error) {
+	// NewDevelopment to use the console encoder.
+	l, _ := zap.NewDevelopment()
+	defer func(l *zap.Logger) {
+		_ = l.Sync()
+	}(l)
+
+	l.Sugar().Fatal(err)
 }
```

Comment:
1) Not using `panic()`.
2) Printing timestamp.
3) Printing stacktraces.

For 3) i'll push a fix as it should be `Fatalf("%+v", err)`.

Reply:
Is printing stack traces even necessary here? The use of the function is limited to the early startup (as also stated by its doc comment), so wouldn't they be something like "main calls whatever it says in the error message" all the time, so that there isn't really any additional information in the stack trace?

For me, seeing a stack trace always screams "the software is broken", so it should only print one in unexpected cases. During the initialization phase, when config is loaded etc., errors due to wrong configuration are to be expected and throwing a stack trace at the user just distracts from the problem. I don't think that the following change in behavior is an improvement:

### main branch
```console
$ ./icingadb --doesnotexist
unknown flag `doesnotexist'
```

### This PR
```console
$ ./icingadb --doesnotexist
unknown flag `doesnotexist'
2024-05-08T12:42:06.441+0200	FATAL	logging/logging.go:135	unknown flag `doesnotexist'
can't parse CLI flags
github.com/icinga/icingadb/pkg/config.ParseFlags
	/home/jbrost/src/icingadb/pkg/config/config.go:63
github.com/icinga/icingadb/internal/command.New
	/home/jbrost/src/icingadb/internal/command/command.go:27
main.run
	/home/jbrost/src/icingadb/cmd/icingadb/main.go:41
main.main
	/home/jbrost/src/icingadb/cmd/icingadb/main.go:37
runtime.main
	/usr/lib/go/src/runtime/proc.go:271
runtime.goexit
	/usr/lib/go/src/runtime/asm_amd64.s:1695
github.com/icinga/icingadb/pkg/logging.Fatal
	/home/jbrost/src/icingadb/pkg/logging/logging.go:135
main.run
	/home/jbrost/src/icingadb/cmd/icingadb/main.go:43
main.main
	/home/jbrost/src/icingadb/cmd/icingadb/main.go:37
runtime.main
	/usr/lib/go/src/runtime/proc.go:271
```

Reply:
Yes, a stack trace is not necessary in this case. This is an oversight. But if `defaults.Set()` in `FromYAMLFile()` fails, or if the argument passed to `FromYAMLFile()`/`ParseFlags()` is invalid, I would like to see one. In such cases, we could also just use `panic()`. What do you think?

For printing all the other errors we have two options:

1) ```can't parse CLI flags: unknown flag `doesnotexist'```

2) ```2024-05-08T13:55:41.738+0200    ERROR    logging/logging.go:138    can't parse CLI flags: unknown flag `doesnotexist'```

Reply:
I prefer 1. If this function is used, it's most likely the only thing that will be printed by this process (logging wasn't initialized, so no previous log messages where you could compare times with). If this happens when the process was manually started in the foreground, you see the error immediately, if it happens from within a systemd unit, journald will track the time of the message already.

Reply:
Done.